src/logics/aave-v3/service.ts

```diff
@@ -132,83 +69,125 @@ export class Service extends common.Web3Toolkit {
+      const reserveMap: Record<string, any> = {};
+
+      const poolAddress = await this.getPoolAddress();
+      const assetAddresses = await Pool__factory.connect(poolAddress, this.provider).getReservesList();
+
+      const calls: common.Multicall3.CallStruct[] = assetAddresses.flatMap((assetAddress) => [
+        {
+          target: this.poolDataProvider.address,
+          callData: this.poolDataProviderIface.encodeFunctionData('getReserveConfigurationData', [assetAddress]),
+        },
+        {
+          target: this.poolDataProvider.address,
+          callData: this.poolDataProviderIface.encodeFunctionData('getReserveTokensAddresses', [assetAddress]),
+        },
+      ]);
+      const { returnData } = await this.multicall3.callStatic.aggregate(calls);
+
+      assetAddresses.forEach((assetAddress, i) => {
+        const { isActive, isFrozen, borrowingEnabled } = this.poolDataProviderIface.decodeFunctionResult(
+          'getReserveConfigurationData',
+          returnData[i * 2]
+        );
+
+        reserveMap[assetAddress] = {
+          isSupplyEnabled: isActive && !isFrozen,
+          isBorrowEnabled: isActive && !isFrozen && borrowingEnabled,
+        };
+
+        const { aTokenAddress, stableDebtTokenAddress, variableDebtTokenAddress } =
+          this.poolDataProviderIface.decodeFunctionResult('getReserveTokensAddresses', returnData[i * 2 + 1]);
+
+        tokenAddresses.push(assetAddress, aTokenAddress, stableDebtTokenAddress, variableDebtTokenAddress);
+      });
+
       const tokens = await this.getTokens(tokenAddresses);
 
-      this.reserveTokens = [];
-      let j = 0;
-      for (let i = 0; i < reserveTokensAddresses.length; i++) {
-        const asset = tokens[j];
-        j++;
-        const aToken = tokens[j];
-        j++;
-        const stableDebtToken = tokens[j];
-        j++;
-        const variableDebtToken = tokens[j];
-        j++;
-        this.reserveTokens.push({ asset, aToken, stableDebtToken, variableDebtToken });
+      for (let i = 0; i < tokens.length; i += 4) {
+        const assetAddress = tokens[i].address;
+
+        reserveMap[assetAddress].asset = tokens[i];
+        reserveMap[assetAddress].aToken = tokens[i + 1];
+        reserveMap[assetAddress].stableDebtToken = tokens[i + 2];
+        reserveMap[assetAddress].variableDebtToken = tokens[i + 3];
       }
+
+      this.reserveTokens = Object.values(reserveMap);
+
+      // Add aToken address as key for quick lookup
+      for (const reserve of Object.values(reserveMap)) {
+        reserveMap[reserve.aToken.address] = reserve;
+      }
+
+      this.reserveMap = reserveMap;
     }
 
-    return this.reserveTokens;
+    return { reserveTokens: this.reserveTokens!, reserveMap: this.reserveMap! };
+  }
+
+  async getSupplyTokens() {
+    const { reserveTokens } = await this.getReserveTokens();
+
+    return reserveTokens.filter(({ isSupplyEnabled, asset }) => isSupplyEnabled && asset.symbol !== 'GHO');
+  }
```

Comment:
I couldn't find an on-chain method to limit the supply of GHO, so I filtered by symbol instead.

Reply:
add the comment for that, and updated only mainnet GHO is not allowed for supply